### PR TITLE
APP-59 SDK part: CC32xx fixes

### PR DESCRIPTION
**Pull request by forGGe**
List of changes as follows:
- build improvements
- fix of crash during initialization in kaa.c
- workaround for fragile persistent. See [KAA-845](http://jira.kaaproject.org/browse/KAA-845) for further details

Check commits for further info.

### CODE REVIEW

**Comment by forGGe** [comment]
cc @dyosick @rasendubi @GamovCoder @RostakaGmfun

File: client/client-multi/client-c/src/kaa/kaa.c [hunk 1]
```diff
@@ -40,71 +40,90 @@ static kaa_error_t kaa_context_create(kaa_context_t **context_p, kaa_logger_t *l
...
+
 
-    if (!error)
+    if (!error){
         error = kaa_bootstrap_manager_create(&context->bootstrap_manager, context);
+    }
+
 
-    if (!error)
+    if (!error){
```
Comment (ohamov) [inline client/client-multi/client-c/src/kaa/kaa.c]: CS

File: client/client-multi/client-c/src/kaa/kaa.c [hunk 2]
```diff
@@ -40,71 +40,90 @@ static kaa_error_t kaa_context_create(kaa_context_t **context_p, kaa_logger_t *l
...
         error = KAA_ERR_NOMEM;
+    }
 
-    if (!error)
+    if (!error) {
         error = kaa_status_create(&context->status->status_instance);
+    }
 
-    if (!error)
+    if (!error){
```
Comment (GamovCoder) [inline client/client-multi/client-c/src/kaa/kaa.c]: CS

**Comment by RostakaGmfun** [comment]
LGTM

**Comment by forGGe** [comment]
Merge conflicts resolved

File: client/client-multi/client-c/src/kaa/platform-impl/cc32xx/cc32xx_status.c [hunk 3]
```diff
@@ -30,3 +30,10 @@ void ext_status_store(const char *buffer, size_t buffer_size)
 {
     cc32xx_binary_file_store(KAA_STATUS_STORAGE, buffer, buffer_size);
 }
+
+void ext_status_delete()
```
Comment (rasendubi) [inline client/client-multi/client-c/src/kaa/platform-impl/cc32xx/cc32xx_status.c]: Shouldn't it be `(void)`?

**Comment by rasendubi** [comment]
Minor nitpick. Otherwise looks good.

**Comment by rasendubi** [comment]
LGTM

**Comment by RostakaGmfun** [comment]
Run build

**Pull request merged**